test/python/transpiler/test_minimum_point.py

```diff
@@ -0,0 +1,181 @@
+        min_pass.property_set["size"] = 20
+        min_pass.run(dag)
+        self.assertEqual(min_pass.property_set["test_minimum_point_count"], 1)
+        self.assertIsNone(min_pass.property_set["test_backtrack_history"])
+        self.assertIsNone(min_pass.property_set["test_min_point"])
+        min_pass.property_set["fidelity"] = 0.775
+        min_pass.property_set["depth"] = 25
+        min_pass.property_set["size"] = 35
+        min_pass.run(dag)
+        self.assertEqual(min_pass.property_set["test_minimum_point_count"], 1)
+        self.assertEqual((0.775, 25, 35), min_pass.property_set["test_backtrack_history"][0])
+        self.assertIsNone(min_pass.property_set["test_min_point"])
+        min_pass.property_set["fidelity"] = 0.775
+        min_pass.property_set["depth"] = 45
+        min_pass.property_set["size"] = 35
+        min_pass.run(dag)
+        self.assertEqual(min_pass.property_set["test_minimum_point_count"], 2)
+        self.assertEqual((0.775, 25, 35), min_pass.property_set["test_backtrack_history"][0])
+        self.assertIsNone(min_pass.property_set["test_minimum_point"])
+        min_pass.property_set["fidelity"] = 0.775
+        min_pass.property_set["depth"] = 36
+        min_pass.property_set["size"] = 40
+        min_pass.run(dag)
+        self.assertEqual(min_pass.property_set["test_minimum_point_count"], 3)
+        self.assertEqual((0.775, 25, 35), min_pass.property_set["test_backtrack_history"][0])
+        self.assertIsNone(min_pass.property_set["test_minimum_point"])
+        min_pass.property_set["fidelity"] = 0.775
+        min_pass.property_set["depth"] = 36
+        min_pass.property_set["size"] = 40
+        min_pass.run(dag)
+        self.assertEqual(min_pass.property_set["test_minimum_point_count"], 4)
+        self.assertEqual((0.775, 25, 35), min_pass.property_set["test_backtrack_history"][0])
+        self.assertIsNone(min_pass.property_set["test_minimum_point"])
+        min_pass.property_set["fidelity"] = 0.775
+        min_pass.property_set["depth"] = 36
+        min_pass.property_set["size"] = 40
+        min_pass.run(dag)
+        self.assertEqual(min_pass.property_set["test_minimum_point_count"], 5)
+        self.assertEqual((0.775, 25, 35), min_pass.property_set["test_backtrack_history"][0])
+        self.assertTrue(min_pass.property_set["test_minimum_point"])
+
+    def test_min_reset_backtrack_range(self):
+        """Test minimum resets backtrack depth."""
+        min_pass = MinimumPoint(["fidelity", "depth", "size"], prefix="test")
+        dag = DAGCircuit()
+        min_pass.property_set["fidelity"] = 0.875
+        min_pass.property_set["depth"] = 15
+        min_pass.property_set["size"] = 20
+        min_pass.run(dag)
+        self.assertEqual(min_pass.property_set["test_minimum_point_count"], 1)
+        self.assertIsNone(min_pass.property_set["test_backtrack_history"])
+        self.assertIsNone(min_pass.property_set["test_min_point"])
```

Comment:
Some of the assertions in these tests feel rather like testing relatively minor implementation details, like the exact objects we expect are tracked with very specific names in the property set.

Leaving that aside, though, the test is quite hard to quickly check what the sequence of values are being set as the properties; for something like this, I think it's maybe easier if we could define the sequence of inserted values and expected values up in single lists at the top of the test so all the context is localised, then loop through them.

Reply:
I left the tests mostly in place like this primarily to make it explicit what is happening in each iteration. I worry the details of the exact logic flow would be lost with a for loop. Especially given the complexity of the logic going on here my thinking was it might be difficult to track and testing it explicitly like this made it easier for me to ensure I got the expected logic. I agree that it does end up being a bit repetitive and a bit too detailed. In https://github.com/Qiskit/qiskit-terra/pull/9612/commits/e2c8cd023d86e30e48b32f28d46ba0db91b5d64d I added a bunch of comments to each iteration to explain the context.